internal/proxy/proxy.go

```diff
@@ -443,7 +443,7 @@ func (c *Client) serveSocketMount(ctx context.Context, s *socketMount) error {
 			return err
 		}
 		// handle the connection in a separate goroutine
-		go func() {
+		go func(cConn net.Conn) {
```

Comment:
flyby: I do not quite understand the reason for this change, the prior version seemed clear and is correct.

If you want to keep this version: prefer to name this variable precisely the same as the outer variable to avoid the possibility of future code changes accessing the "wrong" variable by accident. With different names, that mistake is possible; with the same name, it is impossible, and therefore the preferred form.

Reply:
I misunderstood that the for loop creates a new scope each iteration. I'll revert this.